### DISCUSSION TITLE
Pick & Place of an object

### DISCUSSION THREAD

**Discussion by marleyshan21** [stack-of-tasks/pinocchio#2464] [Q&A]
Hi team!

I am utilizing pinocchio to perform pick and place using a custom URDF based robot.

I was able to simulate picking a rod and computed the torques properly by appending the model onto the end effector.

1. Is this the right way to do it? If yes, How do i detach that part? 
2. Are there other better ways to do pick and place using pinocchio?
3. Is there a code based example for this?- https://github.com/stack-of-tasks/pinocchio/blob/master/doc/d-practical-exercises/1-directgeom.md#12-simple-pick-and-place

Regards

**Comment by jorisv** [comment]
Hello,

> Is this the right way to do it? If yes, How do i detach that part?

I think what you are doing is the simplest way.
You can use two model, one with the object attached to the end effector and another one without.
Then, you only have to switch between these two models when grasping/releasing the object.

> Are there other better ways to do pick and place using pinocchio?

Another way to do it is to use a 6d constraint to join two model (the robot and the manipulated object). But it's harder to setup and we don't have nice example right now. 

> Is there a code based example for this?

I don't know.

**Reply by marleyshan21** [reply]
Oh okay!
I could extend it to 2 models - one with and one without. 

But I would eventually like to extend this. I need to have multiple robots interacting with an object maybe like passing an object between manipulators etc. Having such different models might not make it scalable. 

Do you have any suggestions regarding this?
